src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_pyroguard_emberseer.cpp

```diff
@@ -341,39 +305,18 @@ class npc_blackhand_incarcerator : public CreatureScript
 
     struct npc_blackhand_incarceratorAI : public ScriptedAI
     {
-        npc_blackhand_incarceratorAI(Creature* creature) : ScriptedAI(creature) { }
-
-        void Reset() override
-        {
-            me->SetImmuneToAll(true);
-            if (Creature* Emberseer = me->FindNearestCreature(NPC_PYROGAURD_EMBERSEER, 30.0f, true))
-                Emberseer->AI()->SetData(1, 3);
-        }
-
-        void JustDied(Unit* /*killer*/) override
+        npc_blackhand_incarceratorAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript()) 
         {
-            me->DespawnOrUnsummon(10000);
+            Initialize();
         }
 
-        void SetData(uint32 data, uint32 value) override
+        void Initialize()
         {
-            if (data == 1 && value == 1)
-            {
-                me->SetImmuneToAll(false);
-                me->InterruptSpell(CURRENT_CHANNELED_SPELL);
-                _events.CancelEvent(EVENT_ENCAGED_EMBERSEER);
-            }
-
-            if (data == 1 && value == 2)
-                _events.ScheduleEvent(EVENT_ENCAGED_EMBERSEER, 1s);
+            DoCast(SPELL_ENCAGE_EMBERSEER);
```

Comment:
Never put anything other than assigning class fields in Initialize(). Never cast spells/do strange things in any function called by the constructor.
Spells should be cast in Reset() or JustAppeared()

Reply:
Thank you for the feedback, will not do that in the future.